balloon/src/main/kotlin/com/skydoves/balloon/Balloon.kt

```diff
@@ -806,6 +806,7 @@ public class Balloon private constructor(
   ): Balloon {
     this.setOnBalloonDismissListener {
       if (!destroyed) {
+        builder.onBalloonDismissListener?.onBalloonDismiss()
```

Comment:
Also, the `onBalloonDismissListener` listener is currently being invoked on the call site, so we don't need to handle this.

Reply:
@skydoves dismiss Listener of the previous balloon is not getting invoked before relaying the next balloon because you are replacing the dismiss listener of the previous balloon with initialization of the next ballon and not invoking the previous balloon dismiss listener so after the relaying only the code block in the last balloon dismiss listener is executing.
To eliminate this issue we can call the on-dismiss listener of the previous balloon before replacing it with the new onDismiss listener.